core/Tracker.php

```diff
@@ -114,9 +115,21 @@ public function main(Handler $handler, RequestSet $requestSet)
             $this->init();
 
             if ($this->isPreFlightCorsRequest()) {
+
+
+                //check if cors domain is in the white list
+                $corsDomains = GeneralConfig::getConfigValue('cors_domains');
+                if (!empty($corsDomains)) {
+                    if (in_array($_SERVER['HTTP_ORIGIN'], $corsDomains)) {
+                        $corsDomains = $_SERVER['HTTP_ORIGIN'];
+                    }
+                } else {
+                    $corsDomains = '*';
+                }
```

Comment:
I guess the code would currently produce a warning/error when a request comes from a domain that is not on the cors list. In that case `$corsDomains` is still set the the config value, which is an array.
In addition checking with `in_array` might not be an optimal solution. The `HTTP_ORIGIN` might include `scheme`, `host` and `port`. Depending on the config value those values might not match. e.g. due to missing trailing slash, or ports like `:80` or `:443` included in HTTP_ORIGIN but not in config or wise versa. 
Or the config might simply be a `*`, which also should result in `*`.

Reply:
Also when the "cors_domains" includes a wildcard ("*"), we should use the HTTP_ORIGIN header instead of "*" in the response.
Using the "*" in the response is deprecated and some browser will ignore it.

Just like we do in the CORSHandler class: https://github.com/matomo-org/matomo/blob/4.x-dev/core/API/CORSHandler.php#L35

Reply:
Also, we are already calling the CORSHandler here https://github.com/matomo-org/matomo/blob/4.x-dev/piwik.php#L63 , so the `Access-Control-Allow-Origin` should already been set when the Tracker runs (if I am not missing something here). Thus we could just delete the `Common::sendHeader('Access-Control-Allow-Origin: ' . $corsDomains);` line and it should be fine.